config/config.go

```diff
@@ -66,56 +110,83 @@ func ReadConfigFromFile() (ShipperConfig, error) {
 
 // ShipperConfigFromUnitConfig converts the configuration provided by Agent to the internal
 // configuration object used by the shipper.
-// Currently this just converts the given struct to json and tries to deserialize it into the
-// ShipperConfig struct. This is not the right way to do this, but this gets the build and
-// tests passing again with the new version of elastic-agent-client. Migrating fully to this
-// new config structure is part of the overall agent V2 transition, for more details see
-// https://github.com/elastic/elastic-agent/issues/617.
-func ShipperConfigFromUnitConfig(logLevel client.UnitLogLevel, config *proto.UnitExpectedConfig) (ShipperConfig, error) {
-	jsonConfig, err := config.GetSource().MarshalJSON()
+func ShipperConfigFromUnitConfig(level client.UnitLogLevel, rawConfig *proto.UnitExpectedConfig, grpcConfig ShipperClientConfig) (ShipperRootConfig, error) {
+	cfgObject := DefaultConfig()
+
+	logp.L().Debugf("Got new log level: %s", level.String())
+	logp.SetLevel(ZapFromUnitLogLevel(level))
```

Comment:
This changes the global log level, and the log level here per unit. What you actually want to do is set the log level to the lowest level set by any unit.